src/SayMore/UI/ProjectWindow/ProjectWindow.cs

```diff
@@ -61,9 +63,51 @@ private ProjectWindow()
 			ExceptionHandler.AddDelegate(AudioUtils.HandleGlobalNAudioException);
 
 			InitializeComponent();
+			// ReSharper disable once VirtualMemberCallInConstructor
 			_titleFmt = Text;
 			_menuShowMPlayerDebugWindow.Tag = _menuProject.DropDownItems.IndexOf(_menuShowMPlayerDebugWindow);
 			_menuProject.DropDownItems.Remove(_menuShowMPlayerDebugWindow);
+
+			KeyPreview = true;
+			KeyDown += Project_KeyDown;
+		}
+
+		private static Control FindFocusedControl(Control control)
+		{
+			var container = control as IContainerControl;
+			while (container != null)
+			{
+				control = container.ActiveControl;
+				container = control as IContainerControl;
+			}
+			return control;
+		}
+
+		private void Project_KeyDown(object sender, KeyEventArgs e)
+		{
+			var control = FindFocusedControl((Control)sender);
+
+			if (!_keyDownTypes.Contains(control.GetType().Name)) return;
+			if (e.KeyCode != Keys.Return) return;
+
+			// skip multiline text boxes
+			if (control is TextBox txt)
+			{
+				if (txt.Multiline) return;
+			}
+			
+			// prevent the ding
+			e.SuppressKeyPress = true;
```

Comment:
If you use ProcessDialogKey or ProcessCmdKey, you can just return `true`, and this won't be needed. (Even doing it this way, I would think you could set e.Handled = true, but maybe that doesn't work, since this is just handling the key down event.)